haw-pr.jenkinsfile

```diff
@@ -0,0 +1,104 @@
+                    echo 'Building and testing, using display ' + DISPLAY
+                    script {
+                    	def envvars = [ 
+                    		'-DDISPLAY=localhost'+DISPLAY,
+                            '-DargLine="-DDISPLAY=localhost'+DISPLAY+'"'
+                    	].join(' ')
+                        def options = [
+                            '--batch-mode',
+                            //'--quiet',
+                            '--update-snapshots',
+                            '--show-version',
+                            '-Dtycho.localArtifacts=ignore',
+                            // for adjusting the script only, renable later on:
+                             '-Dmaven.test.failure.ignore',
+                             "--fail-at-end",
+                            '-DWORKSPACE=' + env.WORKSPACE,
+                            '-DexcludeJRE'                               
+                        ].join(' ')
+                        def profiles = [
+                            'buildProduct',
+                            'execute-plugin-tests',
+                            'execute-plugin-ui-tests',
+                            'execute-ecma-tests',
+                            'execute-accesscontrol-tests',
+                            'execute-smoke-tests'
+                         //'execute-hlc-integration-tests'
+                        ].join(',')
+
+                        sh """\
+                            pwd
+                            git log -n 1
+                            npm version
+                        """
+                        sh "mvn ${envvars} clean verify -P${profiles} ${options}"
+                        
+                        // The following calls may be used for testing purposes:
+                        // sh "mvn clean verify ${options}"
+                        // for quick testing:
+                        // sh "mvn -o clean"
+
+						// prints out the target folder, not needed here but useful in other cases
+                        // sh "ls -Ral builds/org.eclipse.n4js.product.build/target/repository/"
+                    }
+                } // end wrap
+            } // end steps
+        } // end stage
+        /* commented out here, may be introduced again, left skeleton here:
+        stage('Deploy') {
+            steps {
+                echo 'Deploying....'
+            }
+        }*/
+    }
+
+    post {
+        always {
+         //   archiveArtifacts allowEmptyArchive: true, artifacts: '**/builds/**/target/products/*.zip'
+         //   archiveArtifacts allowEmptyArchive: true, artifacts: '**/tools/**/target/n4jsc.jar'
+         //   archiveArtifacts allowEmptyArchive: true, artifacts: '**/logs/*.log'
+         //   archiveArtifacts allowEmptyArchive: true, artifacts: '**/tests/**/target/**/*-output.txt'
+
+            junit '**/surefire-reports/**/*.xml'
+            //junit '**/failsafe-reports/**/*.xml'
```

Comment:
Having this line commented out might be a problem. If you are executing failsafe tests above and this line is commented out, the build will be marked as successful (i.e. "green") even if one or more failsafe tests did actually fail.

A short while ago, I noticed this situation in file `eclipse-nightly.jenkinsfile` and wanted to investigate further at the end of my current task, i.e. `GH-1503`.

If you want to test this now: just comment in this line and it will produce an error if no failsafe tests were running and you can safely comment it out, again. Otherwise you know this line is necessary.

In general I think it is unfortunate that we rely on invoking the 'junit' method explicitly in order for test failures to be reported. We should check out configuration of surefire/failsafe at some point.